bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,317 @@
+swagger: "2.0"
+info:
+  description: "Deployment API."
+  version: "1.0.0"
+  title: "Kubeflow Deployment"
+  termsOfService: "https://www.kubeflow.org/"
+  contact:
+    email: "kubeflow-engineering@google.com"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+basePath: "/kfctl/v1"
+tags:
+- name: "deployment"
+  description: "A kubeflow deployment on GCP"
+  externalDocs:
+    description: "Find out more"
+    url: "https://www.kubeflow.org/docs/started/getting-started-gke/"
+schemes:
+- "https"
+paths:
+  /projects/{project}/configs:
+    post:
+      tags:
+        - "config"
+      summary: "Create a new kubeflow config entry on source repo"
+      security:
+        - BearerToken: []
+      operationId: "createConfig"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "body"
+          name: "body"
+          description: "Config needs to be created"
+          required: true
+          schema:
+            $ref: "#/definitions/KfConfig"
+      responses:
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+    get:
+      tags:
+        - "config"
+      summary: "List kubeflow config entries in target project"
+      security:
+        - BearerToken: []
+      operationId: "listConfigs"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+      responses:
+        200:
+          description: "Kubeflow config entries"
+          schema:
+            $ref: "#/definitions/KfConfigEntries"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+  /projects/{project}/configs/{config}/version/{version}/file:
```

Comment:
What is the version resource for?

Is it just fetching the config?

If users want to fetch the config why don't they talk to underlying storage (e.g source repo directly)? Is this just a helper API?

Reply:
This helper API provide read / write of single config file in underlying storage:
1. as of today for source repo there are not such API available.
2. existing edit UI is confusing because it aimed to solve more complicated day-to-day dev use case. 
3. If suggest user change config directly with underlying storage, user experience becomes:
    Create config entry in UI -> setup git and git clone -> edit config -> git push -> apply in UI

With this API we can have experience:
Create config entry in UI -> edit config file in UI -> apply in UI 
Then we provide whole experience within UI, would be much easier for usage like editing [GCP options](https://github.com/kubeflow/kubeflow/blob/master/deployment/gke/deployment_manager_configs/cluster-kubeflow.yaml#L52)